bears/general/URLBear.py

```diff
@@ -96,9 +96,36 @@ def parse_pip_vcs_url(link):
         splitted_schema = splitted_at[splitted_at.index('+') + 1:]
         return splitted_schema
 
+    @staticmethod
+    def process_ignored_list(link_ignore_list):
+        new_ignored_list = []
+
+        for link in link_ignore_list:
+            new_link = link
+
+            if new_link.endswith('/'):
+                suffixes = ['', '**']
+            else:
+                suffixes = ['', '/**']
+
+            if new_link.startswith('//'):
+                prefixes = ['http:', 'https:']
+            else:
+                prefixes = ['']
+
+            for prefix in prefixes:
+                for suffix in suffixes:
+                    new_ignored_list.append(prefix + new_link + suffix)
+
+        return new_ignored_list
+
     @staticmethod
     def extract_links_from_file(file, link_ignore_regex, link_ignore_list):
         link_ignore_regex = re.compile(link_ignore_regex)
+
+        if link_ignore_list:
+            link_ignore_list = URLBear.process_ignored_list(link_ignore_list)
```

Comment:
appending the ignore list only if its empty? Are you sure?

Reply:
On the contrary: if _list_ returns true when the _list_ isn't empty.
This check is necessary to handle situations where the list is None.

Reply:
Only when it is not empty

Reply:
Otherwise it returns an empty list instead of None and it fails unit tests (undesirable behavior).

Reply:
`URLBear.foo -> `self.foo`?